src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.spec.ts

```diff
@@ -1,33 +1,406 @@
+    courseName: 'Test Course',
+    feedbackSessionName: 'test session',
+    instructions: 'Instructions',
+
+    submissionStartTime: { hour: 23, minute: 59 },
+    submissionStartDate: { year: 0, month: 0, day: 0 },
+    submissionEndTime: { hour: 23, minute: 59 },
+    submissionEndDate: { year: 0, month: 0, day: 0 },
+    gracePeriod: 0,
+
+    sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
+    customSessionVisibleTime: { hour: 23, minute: 59 },
+    customSessionVisibleDate: { year: 0, month: 0, day: 0 },
+
+    responseVisibleSetting: ResponseVisibleSetting.CUSTOM,
+    customResponseVisibleTime: { hour: 23, minute: 59 },
+    customResponseVisibleDate: { year: 0, month: 0, day: 0 },
+
+    submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
+    publishStatus: FeedbackSessionPublishStatus.NOT_PUBLISHED,
+
+    isClosingEmailEnabled: true,
+    isPublishedEmailEnabled: true,
+
+    templateSessionName: '',
+
+    isSaving: false,
+    isEditable: false,
+    isDeleting: false,
+    isCopying: false,
+    hasVisibleSettingsPanelExpanded: false,
+    hasEmailSettingsPanelExpanded: false,
+  };
+
   let component: InstructorSessionEditPageComponent;
   let fixture: ComponentFixture<InstructorSessionEditPageComponent>;
+  let feedbackSessionsService: FeedbackSessionsService;
+  let feedbackQuestionsService: FeedbackQuestionsService;
+  let courseService: CourseService;
+  let studentService: StudentService;
+  let instructorService: InstructorService;
+  let navigationService: NavigationService;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       imports: [
+        NgbModule,
+        FormsModule,
+        AjaxLoadingModule,
         RouterTestingModule,
         TeammatesRouterModule,
         HttpClientTestingModule,
         InstructorSessionEditPageModule,
+        BrowserAnimationsModule,
       ],
     })
     .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(InstructorSessionEditPageComponent);
+    courseService = TestBed.inject(CourseService);
+    feedbackSessionsService = TestBed.inject(FeedbackSessionsService);
+    feedbackQuestionsService = TestBed.inject(FeedbackQuestionsService);
+    studentService = TestBed.inject(StudentService);
+    instructorService = TestBed.inject(InstructorService);
+    navigationService = TestBed.inject(NavigationService);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should snap with default fields', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap with feedback session questions', () => {
+    component.questionEditFormModels = [testQuestionEditFormModel1, testQuestionEditFormModel2];
+    component.isLoadingFeedbackSession = false;
+    component.isLoadingFeedbackQuestions = false;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
```

Comment:
Could we also set the component's course ID/course name/feedback session name, to verify that it shows up as intended?